### PR TITLE
Successful restart vote restarts on current map

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

A successful restart vote will revert any pending map change. Also performs the end round task so persistence data gets saved.

## Why It's Good For The Game

Admin request for lowpop hours.

## Changelog

:cl: LT3
admin: Successful restart votes will now restart on the current map
code: End round and persistence data will be saved before executing successful restart vote
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
We've had issues with people getting their buddies to come join the server to reboot it so that they don't have to play Birdshot. This should prevent that.

**Comment by CoiledLamb** [comment]
it's weird, i think this would be a fine change, but the actual reasons for implementing it keep skeeving me out.
like it is misguided in addressing the issues of the map, by just forcing players to play it, whether they enjoy it or not.
this is not your fault, ofc, but it still makes me disagree with the PR because it only solves a symptom caused by actual problems with map design

**Comment by SinfulBliss** [comment]
If a small group of 10 players on lowpop want to play on another map, so they vote to restart, why should it FORCE them to play the same map? Genuinely don't understand this PR. 

It's very hard to get a restart vote to go through since nonvoters are counted as "No," so if they actually get enough votes to go through, why shouldn't they get to make this decision?

**Comment by TheVekter** [comment]
> it's weird, i think this would be a fine change, but the actual reasons for implementing it keep skeeving me out. like it is misguided in addressing the issues of the map, by just forcing players to play it, whether they enjoy it or not. this is not your fault, ofc, but it still makes me disagree with the PR because it only solves a symptom caused by actual problems with map design, instead of fixing those problems

Those problems are being fixed, but I don't think we should be allowing behavior where a player can get 2-3 of his friends to join, call a restart vote, then get the restart just to skip the map they don't like. None of the maps we have in rotation are _so bad_ that there's a valid reason to never want to play on them.

**Comment by ZeroKelvn** [comment]
>None of the maps we have in rotation are so bad that there's a valid reason to never want to play on them.

This doesn't stop a lot of people from fleeing whenever Birdshot is brought into the fold. People avoid that map like the plague, worse thank Kilo ever had it, for reasons I'm not going to try to define here. If the active players legitimately do not want to play on a map, all this does is punish them for wanting to play the game, since it would almost certainly be an immediate shuttle call that wastes a minimum of 40 minutes time.

**Comment by EricZilla** [comment]
I've seen it win the vote, then people who chose not to vote calling for a restart, so idk skill issue

**Comment by wesoda25** [comment]
Big 👎. (many words incoming!)

The game is not designed around dead pop. Completely different rules and dynamics are in play, it's essentially a lawless no-mans-land. Player motivations are different, social structure is nonexistent, and jobs are nothing more than a title. I don't understand what valuable feedback (in reference to maps) could even be obtained from such an odd place. 

More importantly is the matter of player enjoyment. Deadpop rounds can often go on for hours and hours, so a map means a LOT more. In a space where there's rarely ever admins, I think this quirk of restart votes is a GOOD thing: it allows players to at least exert a little control over their experience. I think stripping them of this ability is really lame. Phoning your friends to skip a map is obviously a problem and one that admins have to deal with, but I don't think using restart votes to avoid a map that _the majority of active players on a server do not want to play on_ is a problem.

tldr it's a videogame, people come here to have fun, stop trying to tell them you know better

**Comment by TheVekter** [comment]
> The game is not designed around dead pop. Completely different rules and dynamics are in play, it's essentially a lawless no-mans-land. 

If you're suggesting that we have different rules based on low or normal pop, I would be more than happy to help you draft those rules and present them to the head admins.

> Deadpop rounds can often go on for hours and hours, so a map means a LOT more. In a space where there's rarely ever admins, I think this quirk of restart votes is a GOOD thing: it allows players to at least exert a little control over their experience.

The only issues people have said they have with the map (Engineering is fucky/weird, it's dirty, maints are bad) have the exact same impact on low or high pop rounds. The only thing any of the people who keep doing this have told me is "I don't like Birdshot and I'm not playing on Sybil because it sucks".

> Phoning your friends to skip a map is obviously a problem and one that admins have to deal with, but I don't think using restart votes to avoid a map that the majority of active players on a server do not want to play on is a problem.

You're contradicting yourself here. If the majority of people on the server don't want to play the map, then the person wanting to change it wouldn't need to call his friends to come help him do it.

> This doesn't stop a lot of people from fleeing whenever Birdshot is brought into the fold. People avoid that map like the plague, worse thank Kilo ever had it, for reasons I'm not going to try to define here. If the active players legitimately do not want to play on a map, all this does is punish them for wanting to play the game, since it would almost certainly be an immediate shuttle call that wastes a minimum of 40 minutes time.

Nobody is forcing anyone to play on any map they don't want to. They are more than welcome to go to one of our other servers for a round instead. If they _really_ hate the map as much as they do, then they'll either do that or take a break. I don't much care either way. I'm not catering to people who are so incensed about the idea of having to play a round on _fucking Birdshot_ (the horror!) that they'd actively break the rules to avoid having to do so.

**Comment by Timberpoes** [comment]
Just ahelp if you want a restart to force a map change.

We don't allow players to spawn in, force a shuttle call and kill everyone that tries to stop them because they want a map change.

We also don't allow players to do the same via votes  because it accomplishes much the same outcomes. If a majority of players don't like the map to the point they refuse to play it and will instead force restart the shift, I have a solution. They can play on another of our servers not running that map or just fuck right off to another codebase entirely with maps they will actually play on.

Failing that, we have 24/7 coverage between supportmin pings on Discord, admin pings on Discird and relayed/urgent ahelps. Ask the admins to set the new map and restart the shift for you.

**Comment by Timberpoes** [comment]
"The majority of players" arguments can also fuck right off. Our rules aren't suspended just because a majority of players disagree with them applying that shift.

If this change results in more maps getting lowpop play, then it increases the chance we get useful feedback that can trigger more map changes to better support lowpop play. Because players can't skip them

It's a better outcome than anyone else here is bringing up. Short of headcoders blocking this PR, I am mindful to merge it on that basis alone.

**Comment by SinfulBliss** [comment]
> 
> Failing that, we have 24/7 coverage between supportmin pings on Discord, admin pings on Discird and relayed/urgent ahelps. Ask the admins to set the new map and restart the shift for you.

Even urgent ahelps reporting serious rule violations often go unresponded to during deadpop hours, I’m not sure in what world an admin would come on to restart the round because you ahelped saying you wanted another map.

Let players play where they will enjoy playing, that is why there’s a map vote, it’s the players’ decision, and if for whatever reason the majority want to restart to go somewhere else, that should also be their prerogative as the people actually playing the game.

**Review by Timberpoes — APPROVED** [review]
